src/csync/vio/csync_vio_local_unix.cpp

```diff
@@ -129,22 +130,33 @@ std::unique_ptr<csync_file_stat_t> csync_vio_local_readdir(csync_vio_handle_t *d
   }
 #endif
 
-  return file_stat;
+    if (file_stat->path.isNull())
+        return file_stat;
```

Comment:
This can't happen, can it?

Reply:
It can, if the conversion from the locale failed (invalid utf-8 for example)

Reply:
Humm, this only seems to be possible on Windows if `WideCharToMultiByte` fails in `c_utf8_from_locale`. Maybe the `QTextDecoder::hasFailure()` code path on unix should also return a null QByteArray? Else the rest of the `original_path` codepaths probably won't be used properly, or maybe we should remove `original_path` also on Windows to follow the unix implementation comment:

> Conversion error: since we can't report error from this function, just return the original string.  We take care of invalid utf-8 in SyncEngine::treewalkFile

Reply:
yes that's right, but that's outside of the scope of this patch.
I've tried to keep the old behavior in this respect.
Ideally, we don't need original_name at all if we keep the wrong encoding in the "path".